### PR TITLE
add  `UpdateChannel` method to `Endpoint`

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #3984 

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/endpoint.go [hunk 1]
```diff
@@ -772,6 +772,17 @@ func (endpoint *Endpoint) SetChannelState(state channeltypes.State) error {
 	return endpoint.Counterparty.UpdateClient()
 }
 
+// UpdateChannelState updates a channel
+func (endpoint *Endpoint) UpdateChannel(updater func(channel *channeltypes.Channel)) error {
```
[the remark this answers is not in the document]
Comment (charleenfei) [reply testing/endpoint.go:776]: yea, I can mark it as deprecated, good call :)

File: testing/endpoint.go [hunk 2]
```diff
@@ -772,6 +773,18 @@ func (endpoint *Endpoint) SetChannelState(state channeltypes.State) error {
...
+// UpdateChannel updates the channel associated with the given endpoint. It accepts a
+// closure which takes a channel allowing the caller to modify its fields.
+func (endpoint *Endpoint) UpdateChannel(updater func(channel *channeltypes.Channel)) error {
+	channel := endpoint.GetChannel()
+	updater(&channel)
+	endpoint.SetChannel(channel)
+
+	endpoint.Chain.Coordinator.CommitBlock(endpoint.Chain)
+
+	return endpoint.Counterparty.UpdateClient()
```
Comment (colin-axner) [inline testing/endpoint.go:785]: I think we can remove the updating the counterparty like @DimitrisJim did for `UpdateConnection`?

Comment (crodriguezvega) [reply testing/endpoint.go]: I updated it!

**Review by crodriguezvega — APPROVED** [review]

**Review by chatton — APPROVED** [review]
LGTM! Thanks for picking this one up

**Pull request merged**